subnet/watch.go

```diff
@@ -76,7 +77,7 @@ func (lw *leaseWatcher) reset(leases []Lease) []Event {
 
 		found := false
 		for i, ol := range lw.leases {
-			if ol.Subnet.Equal(nl.Subnet) {
+			if ol.Subnet.Equal(nl.Subnet) && bytes.Compare(ol.Attrs.BackendData, nl.Attrs.BackendData) == 0 {
```

Comment:
Just a thought, I think BackendData is a JSON object, right? So since we use byte compare this condition can also potentially trigger if the backend data marshaled differently for "ol" and "nl" as the marshaled field order may not be guaranteed. Probably not a big deal if the object contains just a single field (VtepMAC) most of the time.

Reply:
I suspect a change in the field order would indicate that there has been a write to the object in etcd, which we likely don't want to suppress passing that change to the backend when re-creating the watch anyways. So it's a good point, but I think in this case the bytes.Compare reinforces the expected behaviour of passing along any change notification from etcd to the backend.